### ISSUE TITLE
Please consider removing postinstall scripts

### ISSUE THREAD

**Issue by lensbart** [FortAwesome/Font-Awesome#18803]
Hello,

First of all, thank you for providing the world with great and affordable icons!

`@fortawesome/fontawesome-common-types`, `@fortawesome/fontawesome-svg-core`, `@fortawesome/free-brands-svg-icons` and `@fortawesome/free-solid-svg-icons` specify a `postinstall` script which displays some attribution in the console. The presence of a postinstall script is inconvenient:

* The default behaviour of [Yarn PnP](https://next.yarnpkg.com/features/pnp/) is to not compress these packages, and to include them wholesale in any consuming repository (7MB on disk). This is checked in to version control and included in every deploy.
* It’s possible to override this behaviour by specifying the following in `package.json`:
  ```
  "dependenciesMeta": {
    "@fortawesome/fontawesome-common-types": {
      "built": false
    },
    "@fortawesome/fontawesome-svg-core": {
      "built": false
    },
    "@fortawesome/free-brands-svg-icons": {
      "built": false
    },
    "@fortawesome/free-solid-svg-icons": {
      "built": false
    },
  },
  ```
  but this leads to the following warnings to be displayed by Yarn:
  ```
  ➤ YN0005: │ @fortawesome/fontawesome-svg-core@npm:1.3.0 lists build scripts, but its build has been explicitly disabled through configuration.
  ➤ YN0005: │ @fortawesome/fontawesome-common-types@npm:0.3.0 lists build scripts, but its build has been explicitly disabled through configuration.
  ➤ YN0005: │ @fortawesome/free-brands-svg-icons@npm:6.0.0 lists build scripts, but its build has been explicitly disabled through configuration.
  ➤ YN0005: │ @fortawesome/free-solid-svg-icons@npm:6.0.0 lists build scripts, but its build has been explicitly disabled through configuration.
  ```
* It’s possible to disable these warning by adding the following to `.yarnrc.yml`:
  ```
  logFilters:
    - code: YN0005
      level: discard
  ```
  but this would disable all such warnings, i.e. for other packages as well.

In other words, no convenient option exists for Font Awesome customers who want a distraction-free, minimum-disk-size experience.

In addition, some users seem to run into issues that are more inconvenient:
- https://github.com/FortAwesome/Font-Awesome/issues/18571
- https://github.com/FortAwesome/Font-Awesome/issues/18377
- https://github.com/FortAwesome/Font-Awesome/issues/18005
- https://github.com/FortAwesome/Font-Awesome/issues/17816

All of these would be resolved by removing the "postinstall".

A better explanation of why to avoid postinstall can be found here: https://yarnpkg.com/advanced/lifecycle-scripts/#a-note-about-postinstall

Thank you!

**Comment by tagliala** [comment]
Hi!

Thanks for being part of the Font Awesome Community and thanks for reporting this.

@robmadole could you please take a look?

**Comment by robmadole** [comment]
Thanks, @lensbart we'll take a look.

**Comment by MMana76** [comment]
I have same problem with my Gitlab pipeline.
My Docker logs
```
46798 verbose stack Error: @fortawesome/free-solid-svg-icons@5.15.1 postinstall: `node attribution.js`
46798 verbose stack Exit status 1
46798 verbose stack     at EventEmitter.<anonymous> (/usr/local/lib/node_modules/npm/node_modules/npm-lifecycle/index.js:332:16)
46798 verbose stack     at EventEmitter.emit (events.js:400:28)
46798 verbose stack     at ChildProcess.<anonymous> (/usr/local/lib/node_modules/npm/node_modules/npm-lifecycle/lib/spawn.js:55:14)
46798 verbose stack     at ChildProcess.emit (events.js:400:28)
46798 verbose stack     at maybeClose (internal/child_process.js:1058:16)
46798 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:293:5)
46799 verbose pkgid @fortawesome/free-solid-svg-icons@5.15.1
46800 verbose cwd /var/www/html/fluffa.quantox.dev/fe
46801 verbose Linux 4.9.0-18-amd64
46802 verbose argv "/usr/local/bin/node" "/usr/local/bin/npm" "install"
46803 verbose node v14.19.0
46804 verbose npm  v6.14.16
```

**Comment by Tostis** [comment]
I've the same error due to Docker Desktop that changes the default wsl distro.
The command
`
bash -c node attribution.js
`
doesn't work.
If you call bash from dos prompt it says
`
An error occurred mounting one of your file systems. Please run 'dmesg' for more details.
`
So I listed the wsl distros
`wsl --list  
Windows Subsystem for Linux Distributions:  
docker-desktop-data (Default)  
docker-desktop  
`
and changed the default one
`
wsl -s docker-desktop
`

**Comment by lensbart** [comment]
@tagliala in the [linked issue](https://github.com/FortAwesome/Font-Awesome/issues/19243#issuecomment-1206689375) you said:
> I think this issue is the same as https://github.com/FortAwesome/Font-Awesome/issues/18803, and the fix is planned for 6.2.0

However this doesn’t seem to have been fixed in 6.2?

> Thanks, @lensbart we'll take a look.

Thanks @robmadole — any chance this will get fixed?

**Comment by dbaynard** [comment]
This is still present in 6.4, for multiple packages. See [@fortawesome/fontawesome-common-types - npm Package Overview - Socket](https://socket.dev/npm/package/@fortawesome/fontawesome-common-types), for example. Considering all this example does is a `console.log`, it seems like an enormous cost to pay.

**Comment by lensbart** [comment]
Agreed @dbaynard — very frustrating considering that this impacts paying customers.

@robmadole any update?

**Comment by cwilsn** [comment]
We started getting the same error that was breaking our builds in Laravel Forge (as well as Envoyer).

```
error /home/forge/my-web-app/releases/20240101010101/node_modules/@fortawesome/pro-solid-svg-icons: Command failed.
Exit code: 1
Command: node attribution.js
Arguments: 
Directory: /home/forge/my-web-app/releases/20240101010101/node_modules/@fortawesome/pro-solid-svg-icons
Output:
node:internal/modules/cjs/loader:1137
  throw err;
```

We are on version 6.4.  Node v18, yarn v1

**Comment by lensbart** [comment]
In the meantime I’ve migrated to https://phosphoricons.com, which comes with a less restrictive licence and doesn’t have this issue.

**Comment by robmadole** [comment]
Alrighty, this has gotten pretty annoying for folks. We'll completely remove this in the next version. 😑

**Comment by mbellamyy** [comment]
The problem persists in `6.5.2`... or `6.3.0`... 
node `18.15.0`, yarn `1.22.19`, on WSL Ubuntu 22.04. The workaround doesn't help. Here's the error: 

```error /home/XXXXXXXX/node_modules/@fortawesome/sharp-thin-svg-icons: Command failed.
Exit code: 1
Command: node attribution.js
Arguments: 
Directory: /home/XXXXXXXX/node_modules/@fortawesome/sharp-thin-svg-icons
Output:
node:internal/modules/cjs/loader:1078
  throw err;
  ^

Error: Cannot find module '/home/XXXXXXXX/node_modules/@fortawesome/sharp-thin-svg-icons/attribution.js'
    at Module._resolveFilename (node:internal/modules/cjs/loader:1075:15)
    at Module._load (node:internal/modules/cjs/loader:920:27)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
    at node:internal/main/run_main_module:23:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
```

I've already wasted quite a few hours trying to fix this. It started happening right after I reinstalled my WSL, and I don't have the  same issue on another device running the same WSL version. We're pro subscribers.

Any ideas?

@robmadole sorry for the ping, but it seems like you're the only member who has worked on this issue.

**Comment by GlobeProtocolNick** [comment]
Seems to work for me, after upgrading to version `6.6.0`

**Comment by tagliala** [comment]
Yes, this has been fixed in 6.6.0

#19578
